sidecar/sidecar.go

```diff
@@ -220,5 +232,126 @@ func startProxy(conf *config.Config) error {
 		}
 	}()
 
+	debugger := api.NewDebugAPI(nginxProxy)
+
+	a := rest.NewApi()
+	a.Use(
+		&rest.TimerMiddleware{},
+		&rest.RecorderMiddleware{},
+		&rest.RecoverMiddleware{
+			EnableResponseStackTrace: false,
+		},
+		&rest.ContentTypeCheckerMiddleware{},
+		//&middleware.RequestIDMiddleware{},
+		//&middleware.LoggingMiddleware{},
+		//middleware.NewRequireHTTPS(middleware.CheckRequest{
+		//	IsSecure: middleware.IsUsingSecureConnection,
+		//	Disabled: !conf.RequireHTTPS,
+		//}),
+	)
+
+	routes := debugger.Routes()
+	router, err := rest.MakeRouter(
+		routes...,
+	)
+	if err != nil {
+		logrus.WithError(err).Error("Could not start API server")
+		return err
+	}
+	a.SetApp(router)
+
+	go func() {
+		http.ListenAndServe(fmt.Sprintf(":%v", 6116), a.MakeHandler())
+	}()
+
 	return nil
 }
+
+// Instance TODO
+type Instance struct {
+	Tags string `json:"tags"`
+	Type string `json:"type"`
+	Host string `json:"host"`
+	Port int    `json:"port"`
+}
+
+// NGINXState nginx cached lua state
+type NGINXState struct {
+	Routes    map[string][]rules.Rule `json:"routes"`
+	Instances map[string][]Instance   `json:"instances"`
+	Actions   map[string][]rules.Rule `json:"actions"`
+}
+
+func cliCommand(command string) {
+
+	switch command {
+	case "show-state":
+		httpClient := http.Client{
+			Timeout: time.Second * 10,
+		}
+		req, err := http.NewRequest("GET", "http://localhost:6116/state", nil)
```

Comment:
externalizing the URL is always a better practice, rather than hardcoding it. If one were to change the nginx admin port from 5813 to some other port (as they have another app listening on that port), they would have to recompile the sidecar as well, just to be able to get A8 working (and the debug stuff enabled).

Secondly, it would allow us to remotely debug the sidecar and nginx, if the user decides to expose these two admin ports outside of the container. The user would have to just use the stock a8sidecar binary and point to these two urls, and they can obtain all the internal state inside these entities.